src/compiler/crystal/codegen/debug.cr

```diff
@@ -2,15 +2,22 @@ require "./codegen"
 
 module Crystal
   class CodeGenVisitor
-    CRYSTAL_LANG_DEBUG_IDENTIFIER = 0x8002_u32
+    CRYSTAL_LANG_DEBUG_IDENTIFIER = 0x28_u32
+    #
+    # We have to use it because LLDB has builtin type system support for C++/clang that we can use for now for free.
+    # Later on we can implement LLDB Crystal type system so we can get official Language ID
+    #
+    CPP_LANG_DEBUG_IDENTIFIER = 0x0004_u32
 
     @current_debug_location : Location?
+    @debug_files = {} of Crystal::VirtualFile | String? => LibLLVMExt::Metadata
+    @current_debug_file : LibLLVMExt::Metadata?
 
     def di_builder(llvm_module = @llvm_mod || @main_mod)
       di_builders = @di_builders ||= {} of LLVM::Module => LLVM::DIBuilder
       di_builders[llvm_module] ||= LLVM::DIBuilder.new(llvm_module).tap do |di_builder|
         file, dir = file_and_dir(llvm_module.name == "" ? "main" : llvm_module.name)
-        di_builder.create_compile_unit(CRYSTAL_LANG_DEBUG_IDENTIFIER, file, dir, "Crystal", 0, "", 0_u32)
+        di_builder.create_compile_unit(CPP_LANG_DEBUG_IDENTIFIER, file, dir, "Crystal", !@debug.variables?, "", 0_u32)
```

Comment:
What does it mean here that we do `!@debug.variables?`. This is a bit confusing to me. `@debug` is a flags enum with these possible flags: `LineNumbers`, `Variables`. So this means the value we pass here is `true` if the value is `None` or `LineNumbers`... is that correct?

Maybe add a comment explaining why we are passing that value?

Reply:
@debug.variables? is set to true if parameter --debug is set in command line.
This flag affects only debug variables generation. It sets Optimized parameter to false.

Reply:
I just copied my comment above as a comment into the code.

Reply:
Perhaps it would be more descriptive to do:

```suggestion
        is_optimized = !@debug.variables?
        di_builder.create_compile_unit(CPP_LANG_DEBUG_IDENTIFIER, file, dir, "Crystal", is_optimized, "", 0_u32)
```